pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,334 @@
+            if k.startswith('scene-'):
+                configs = self.cfg.getliteral(cfgsect, k)
+                field = configs['field']
+                sname = k.removeprefix('scene-')
+
+                scene_cfg[sname] = configs
+
+                for kc, vc in configs.items():
+                    if kc == 'image_name':
+                        pass
+                    elif kc == 'field_expr':
+                        if field in self._expr:
+                            raise KeyError(f"Field '{field}' already exists")
+                        self._expr[field] = self._get_expr(vc, subs=cons)
+                    elif kc.startswith('field_expr_'):
+                        sub_field = kc.removeprefix('field_expr_')
+                        if sub_field not in ['x', 'y', 'z'][:self.ndims]:
+                            raise KeyError('Invalid vector subfield ID')
+                        if field not in self._expr:
+                            self._expr[field] = {}
+                        elif not isinstance(self._expr[field], dict):
+                            raise KeyError(f"Field  '{field}' already exists")
+                        expr = self._get_expr(vc, subs=cons)
+                        self._expr[field][sub_field] = expr
+                    elif kc == 'pipeline':
+                        self.scenes[f's{i}/plots/p1/pipeline'] = f'pl_{vc}'
+                    elif kc == 'field':
+                        self.scenes[f's{i}/plots/p1/{kc}'] = field
+                    elif kc.startswith('render-'):
+                        ropt = kc.removeprefix('render-')
+                        self.scenes[f's{i}/renders/r1/{ropt}'] = vc
+                    else:
+                        self.scenes[f's{i}/plots/p1/{kc}'] = vc
+                i += 1
+        self._scene_cfg = scene_cfg
+
+    def _config_pipeline(self, cfgsect):
+        pipel_cfg = {}
+        self.pipelines = pl = ConduitNode(self.conduit)
+
+        for k in self.cfg.items(cfgsect):
+            if k.startswith('pipeline-'):
+                configs = self.cfg.getliteral(cfgsect, k)
+                configs = [configs] if isinstance(configs, dict) else configs
+                pname = k.removeprefix('pipeline-')
+
+                pipel_cfg[pname] = configs
+                params = ConduitNode(self.conduit)
+                for j, filter in enumerate(configs):
+                    for kf, vf in filter.items():
+                        if kf == 'type':
+                            ptype = vf
+                            pl[f'pl_{pname}/f{j}/{kf}'] = vf
+                        else:
+                            params[kf] = vf
+
+                        if kf == 'field' and vf not in self._expr:
+                            raise KeyError(f"Missing field '{vf}' used in pipeline")
+                if ptype in self.v_filter and 'velocity' not in self._expr:
+                    params['field'] = 'velocity'
+                    v = {k: v for k, v in zip('xyz'[:2], 'uvw')}
```

Comment:
Fragile, assumes all systems will use u, v, w for their velocity field names

Reply:
Ok, this can't be implicitly handled then and instead the field setting will have to be used for users to manually set the velocity vector